adapters/roulax/roulax.go

```diff
@@ -0,0 +1,144 @@
+package roulax
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+
+type ExtImpRoulax struct {
+	Pid   string `json:"Pid,omitempty"`
+	PublisherPath string `json:"publisherPath,omitempty"`
+}
+
+
+
+// Builder builds a new instance of the Adot adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	fmt.Println(config.Endpoint)
+	return bidder, nil
+}
+
+
+
+// getImpAdotExt parses and return first imp ext or nil
+func getImpRoulaxExt(imp *openrtb2.Imp) *ExtImpRoulax {
+	var extImpRoulax ExtImpRoulax
+	var extBidder adapters.ExtImpBidder
+	err := json.Unmarshal(imp.Ext, &extBidder)
+	if err != nil {
+		return nil
+	}
+	err = json.Unmarshal(extBidder.Bidder, &extImpRoulax)
+	if err != nil {
+		return nil
+	}
+	return &extImpRoulax
+}
+
+// MakeRequests makes the HTTP requests which should be made to fetch bids.
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	reqJson, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+
+	roulaxExt := getImpRoulaxExt(&request.Imp[0])
```

Comment:
Any reason of using just one impression and ignoring others. There could be multiple impresssions in the request. Would you like to create a separate request for each imp?